### DISCUSSION TITLE
Bug / regression in livewire datatable pagination introduced by a livewire commit

### DISCUSSION THREAD

**Discussion by marionfromfrance** [livewire/livewire#3996] [7. Bugs (archived)] [answered]
Hello everyone,
I had an issue with livewire datatables and git bisect here : https://github.com/MedicOneSystems/livewire-datatables/issues/315. 
This led me to this place, because my git bisect said this bug was introduced by : 

52a8605efe48a3900092aac67179531564113eff is the first bad commit
commit 52a8605efe48a3900092aac67179531564113eff
Author: Josh
Date:   Fri Sep 10 03:31:04 2021 +1000

    [Feature] multiple paginators support (#3684)
    
    * Add two paginators test
    
    * Add support for two paginators
    
    * Initialise page in paginators array by default
    
    * Fix formatting
    
    * Add hook support for multiple paginators

I can help to fix it, but any help is welcome :)

When I replace : 
`Paginator::currentPageResolver(function ($pageName) {
            if (! isset($this->paginators[$pageName])) {
                $this->paginators[$pageName] = request()->query($pageName, 1);
            }
            return (int) $this->paginators[$pageName];
        });`

with 
`Paginator::currentPageResolver(function () {
            return (int) $this->page;});`

it works again

**Comment by marionfromfrance** [comment, marked as answer]
I made a contribution to fix this issue in livewire datatable because $this->page was used instead of $this->setPage() call. But maybe the $page attribute could be private in order to avoid this issue with libraries?
Here is the pull request : https://github.com/MedicOneSystems/livewire-datatables/pull/319

**Reply by joshhanley** [reply]
@marionfromfrance ah good find! Thanks for being willing to contribute and fix that package up 🙂

We had to make the change to support multiple paginators, but yeah if they're not using the pagination methods like `setPage` that'd explain why it broke (not ideal that it broke though). 

Yeah ideally people shouldn't be able to modify the `$page` attribute directly, but as it's included as a trait into the component, then technically they can do whatever they like with it as we have no way of hiding it.

Hope this helps!
